plugins/node/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -0,0 +1,749 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  context,
+  diag,
+  propagation,
+  trace,
+  Span,
+  SpanKind,
+  SpanStatusCode,
+} from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import {
+  SemanticAttributes,
+  MessagingOperationValues,
+  MessagingDestinationKindValues,
+} from '@opentelemetry/semantic-conventions';
+import type * as amqp from 'amqplib';
+import {
+  AmqplibInstrumentationConfig,
+  DEFAULT_CONFIG,
+  EndOperation,
+} from './types';
+import {
+  CHANNEL_CONSUME_TIMEOUT_TIMER,
+  CHANNEL_SPANS_NOT_ENDED,
+  CONNECTION_ATTRIBUTES,
+  getConnectionAttributesFromServer,
+  getConnectionAttributesFromUrl,
+  InstrumentationConsumeChannel,
+  InstrumentationMessage,
+  InstrumentationPublishChannel,
+  isConfirmChannelTracing,
+  markConfirmChannelTracing,
+  MESSAGE_STORED_SPAN,
+  normalizeExchange,
+  unmarkConfirmChannelTracing,
+} from './utils';
+import { VERSION } from './version';
+
+export class AmqplibInstrumentation extends InstrumentationBase<typeof amqp> {
+  protected override _config!: AmqplibInstrumentationConfig;
+
+  constructor(config: AmqplibInstrumentationConfig = {}) {
```

Comment:
```suggestion
  constructor(config?: AmqplibInstrumentationConfig) {
```

Reply:
Looks like there are many instrumentations using the default argument pattern and many that use the optional param pattern.
When should we use each?

Reply:
A while ago I updated the core repo in this regard (see https://github.com/open-telemetry/opentelemetry-js/pull/2522) but I never found the time to do the same in contrib.

There are no rules as far as I know it's just not needed to use a default here.

Reply:
cool, I changed it in amqplib 👍